### PR TITLE
Removed implicit conversion warnings

### PR DESCRIPTION
**Pull request by fredevb**
Fixed some implicit conversion warnings.

### CODE REVIEW

File: core/include/traccc/seeding/spacepoint_binning_helper.hpp [hunk 1]
```diff
@@ -82,7 +82,7 @@ inline std::pair<detray::axis2::circular<>, detray::axis2::regular<>> get_axes(
 
         // divide 2pi by angle delta to get number of phi-bins
         // size is always 2pi even for regions of interest
-        phiBins = std::ceil(2 * M_PI / deltaPhi);
+        phiBins = static_cast<detray::index>(std::ceil(2 * M_PI / deltaPhi));
```
Comment (stephenswat) [inline core/include/traccc/seeding/spacepoint_binning_helper.hpp:85]: This should be `std::llround(2 * M_PI / deltaPhi + 0.5)`.

File: core/src/ambiguity_resolution/greedy_ambiguity_resolution_algorithm.cpp [hunk 2]
```diff
@@ -245,7 +245,7 @@ void greedy_ambiguity_resolution_algorithm::compute_initial_state(
             double ratio = static_cast<float>(mcount_idzero) /
                            static_cast<float>(mcount_all);
             if (ratio > warning_threshold) {
-                std::size_t percent = ratio * 100;
+                std::size_t percent = static_cast<std::size_t>(ratio * 100);
```
Comment (stephenswat) [inline core/src/ambiguity_resolution/greedy_ambiguity_resolution_algorithm.cpp:248]: Let's just render this as a floating point number. E.g.:

```cpp
std::stringstream stream;
stream << std::fixed << std::setprecision(2) << (ratio * 100.);
std::string s = stream.str();
```

File: core/include/traccc/seeding/spacepoint_binning_helper.hpp [hunk 3]
```diff
@@ -116,8 +116,8 @@ inline TRACCC_HOST_DEVICE size_t is_valid_sp(const seedfinder_config& config,
     if (spPhi > config.phiMax || spPhi < config.phiMin) {
         return detray::detail::invalid_value<size_t>();
     }
-    size_t r_index = getter::perp(
-        vector2{sp.x() - config.beamPos[0], sp.y() - config.beamPos[1]});
+    size_t r_index = static_cast<size_t>(getter::perp(
+        vector2{sp.x() - config.beamPos[0], sp.y() - config.beamPos[1]}));
```
Comment (stephenswat) [inline core/include/traccc/seeding/spacepoint_binning_helper.hpp:120]: Casting this directly to an integer seems wrong, unless the bin size is exactly a unit of length 1... Are you sure this is correct?

**Comment by stephenswat** [comment]
Also, please work on more descriptive PR titles and descriptions.

File: core/include/traccc/seeding/spacepoint_binning_helper.hpp (hunk 3 above)
Comment (fredevb) [reply core/include/traccc/seeding/spacepoint_binning_helper.hpp:120]: I was just fixing the warning:
`implicit conversion turns floating-point number into integer: 'float' to 'size_t' (aka 'unsigned long') [-Werror,-Wfloat-conversion] 
size_t r_index = getter::perp(`

Comment (stephenswat) [reply core/include/traccc/seeding/spacepoint_binning_helper.hpp:120]: I'm suspicious of this line... I would have expected a division by the size of each r-bin in order to get an actual index. Again, this code only works if the bin size is exactly one. But I can open an issue regarding this.

**Pull request closed without merging**